### ISSUE TITLE
Disable blueman-applet in Gnome

### ISSUE THREAD

**Issue by ispanos** [blueman-project/blueman#2102]
blueman: blueman.x86_64 -- 1:2.3.5-3.fc38
BlueZ: bluez.x86_64 -- 5.68-1.fc38
Distribution: Fedora 38
Desktop environment: Gnome / Swaywm

I use Blueman in Sway and I start it in the config file. However, when using Gnome, blueman applet auto-starts and I can't find a way to disable it. In case it is relevant, here is the output of `systemctl --user list-units | grep blue`

```
  sys-devices-pci0000:00-0000:00:14.0-usb3-3\x2d10-3\x2d10:1.0-bluetooth-hci0-hci0:3585.device loaded active plugged /sys/devices/pci0000:00/0000:00:14.0/usb3/3-10/3-10:1.0/bluetooth/hci0/hci0:3585
  sys-devices-pci0000:00-0000:00:14.0-usb3-3\x2d10-3\x2d10:1.0-bluetooth-hci0.device           loaded active plugged /sys/devices/pci0000:00/0000:00:14.0/usb3/3-10/3-10:1.0/bluetooth/hci0
  sys-subsystem-bluetooth-devices-hci0.device                                                  loaded active plugged /sys/subsystem/bluetooth/devices/hci0
  sys-subsystem-bluetooth-devices-hci0:3585.device                                             loaded active plugged /sys/subsystem/bluetooth/devices/hci0:3585
  app-gnome-blueman-17533.scope                                                                loaded active running Application launched by gnome-session-binary
  bluetooth.target                                                                             loaded active active  Bluetooth
```

Thank you for your time.

P.S. I have [disabled bluetooth on boot](https://github.com/blueman-project/blueman/wiki/Troubleshooting#turn-on-bluetooth-device-on-boot) and I can't figure out how to enable using a key-bind. `bluetoothctl power on` works only after I first enable bluetooth using blueman or gnome settings. I would appreciate it if you could point me to the right direction.

**Comment by cschramm** [comment]
Gnome seems to use an application called Tweaks to manage "Startup Applications". You should find and have an option to disable blueman there.

Could you elaborate on "`bluetoothctl power on` does not work"?

**Comment by ispanos** [comment]
> You should find and have an option to disable blueman there.

It is not in tweaks nor in any of the directories for autostart (~/.config/autostart $XDG_CONFIG_HOME/autostart   $XDG_CONFIG_DIRS/autostart   /usr/share/gnome/autostart )

```
$ systemctl --user disable blueman-applet
The unit files have no installation config (WantedBy=, RequiredBy=, Also=,
Alias= settings in the [Install] section, and DefaultInstance= for template
units). This means they are not meant to be enabled or disabled using systemctl.

Possible reasons for having this kind of units are:
• A unit may be statically enabled by being symlinked from another unit's
  .wants/ or .requires/ directory.
• A unit's purpose may be to act as a helper for some other unit which has
  a requirement dependency on it.
• A unit may be started when needed via activation (socket, path, timer,
  D-Bus, udev, scripted systemctl call, ...).
• In case of template units, the unit is meant to be enabled with some
  instance name specified.
```

> Could you elaborate on "bluetoothctl power on does not work"?

I just restarted my laptop so I can share the output in the terminal. I have pasted some relevant information bellow. After reboot due to [disabling bluetooth on boot](https://github.com/blueman-project/blueman/wiki/Troubleshooting#turn-on-bluetooth-device-on-boot) I use blueman applet to enable it. For some reason I had totally forgotten about `rfkill`.

```
$ bluetoothctl power on
No default controller available
$ sudo bluetoothctl power on
No default controller available
$ rfkill list
0: tpacpi_bluetooth_sw: Bluetooth
	Soft blocked: yes
	Hard blocked: no
2: phy0: Wireless LAN
	Soft blocked: no
	Hard blocked: no

$ rfkill unblock 0
0: tpacpi_bluetooth_sw: Bluetooth
	Soft blocked: no
	Hard blocked: no
2: phy0: Wireless LAN
	Soft blocked: no
	Hard blocked: no
3: hci0: Bluetooth
	Soft blocked: no
	Hard blocked: no
$ bluetoothctl power on
[CHG] Controller A0:E7:0B:2A:41:48 PowerState: off-enabling
[CHG] Controller A0:E7:0B:2A:41:48 Class: 0x007c010c (8126732)
Changing power on succeeded
```


I want to turn on bluetooth  to [connect to my headset](https://github.com/ispanos/dotfiles/blob/6039480cd0340af26ac5472ae7b8fd933faa65f4/.local/bin/btoggle#L76). Should I use `rfkill` or is there  more appropriate way? (I need to be able to do it without sudo privileges.)

**Comment by cschramm** [comment]
I don't exactly know about the Fedora package, but generally blueman ships a desktop file, usually installed to `/etc/xdg/autostart/blueman.desktop`. Desktops pick that up for autostart by default and provide a way to disable it. Gnome's way, to my understanding, is Tweaks.

If a killswitch is enabled, `rfkill` is required to first get the adapter available at all, yes.

**Comment by ispanos** [comment]
> /etc/xdg/autostart/blueman.desktop

Indeed this file exists. However there are also other dekstop files in that directory, e.g. nm-applet.desktop, that do not start automatically. Just to clarify, the only program that appears in gnome-tweaks  autostart is dropbox, which I added manually.

**Comment by cschramm** [comment]
`nm-applet.desktop` typically has `NotShowIn=KDE;GNOME;`, so that Gnome ignores it. I have no idea why Tweaks does not pick blueman. Just tried a live system and actually had no autostart entries at all. No idea how to manage things in Gnome then. It used to have a tool called `gnome-session-settings` but that is long gone. Last resort is, of course, to just remove the file. :man_shrugging:

**Comment by cschramm** [comment]
I do not really get the use case, to be honest. Why would a Gnome user even install blueman if he does not intend to use it?

The killswitch behavior makes sense. blueman-applet actually has a "KillSwitch" plugin that does exactly that
> Switches Bluetooth killswitch status to match Bluetooth power state.

**Comment by ispanos** [comment]
> I do not really get the use case, to be honest.

On my main laptop I have been using Swaywm  for more than a year, but I will be using Gnome for some time. Thus I need blueman installed for sway, but not running when I use gnome. Also, on our share system I use Sway and my gf uses Gnome. I don't share my main laptop, but not all systems are used by a single user. It should be possible for each user do choose his/her autostart-applications.

> Switches Bluetooth killswitch status to match Bluetooth power state.

Is it the same as using `rfkill block` ? 

> blueman-applet actually has a "KillSwitch" plugin that does exactly that

Does that reduce power consumption?

**Comment by cschramm** [comment]
> I need blueman installed for sway, but not running when I use gnome.

That sounds rather specific and arbitrary. One could also want to start blueman on Gnome (his primary desktop) but not on another (secondary) desktop. I don't see why we should exclude it from autostart in Gnome and think autostart for all systems (that do support /etc/xdg/autostart) is a sane default. I'm sure there is a way to configure autostart on current Gnome like on any other system, although I do not see it.

> > Switches Bluetooth killswitch status to match Bluetooth power state.
> 
> Is it the same as using `rfkill block` ?

Basically yes. It blocks the kill switch when Bluetooth is powered off and unblocks it when it's powered on. ("Bluetooth is powered" is a little bit misleading as it's actually per adapter, but let's assume just a single one.)

> > blueman-applet actually has a "KillSwitch" plugin that does exactly that
> 
> Does that reduce power consumption?

I would not expect it to have a relevant impact. I guess the motivation for the plugin was mostly to reduce rfkill and the concept of powering adapters to just a single Bluetooth on / off state.

**Comment by ispanos** [comment]
> That sounds rather specific and arbitrary. One could also want to start blueman on Gnome (his primary desktop) but not on another (secondary) desktop.

Most Sway users I know switch to Gnome when they want to game, as there significant performance gains for many games. But that is not the most important reason.

`/etc/xdg/autostart` doesn't work by default in Fedora workstation and is system wide. `NotShowIn=GNOME;` would stop auto-starting for all users and anyone can add with gnome tweaks it if its needed. It would also be better for systems with multiple users.

**Comment by cschramm** [comment]
That's not an argument for excluding Gnome, you're arguing for not shipping an autostart desktop file at all.

The standard way to disable the system-wide default is to place a user-specific file that overrides it. That'd typically go to  `~/.config/blueman.desktop` and you can completely disable anything by adding `Hidden=true` to yours. Other options are `NotShowIn`, `OnlyShowIn` etc.

**Comment by ispanos** [comment]
I am not sure what should happen because this is the first time I'm dealing with something like that.

Is that directory in `/etc/xdg/autostart` the proper place for blueman? I'm not asking for proof, if you are sure a yes would suffice. I'm just wondering because the rest of the desktop files there are not things that  user should interact with. A tray icon, especially when for some DE it is redundant, should be easy to disable. Tray icons are not even officially supported in Gnome.

> The standard way to disable the system-wide default is to place a user-specific file that overrides it. That'd typically go to ~/.config/blueman.desktop and you can completely disable anything by adding Hidden=true to yours. Other options are NotShowIn, OnlyShowIn etc.

If you keep things as is I will solve my issue by adding the file you suggest in `$XDG_CONFIG_HOME/autostart/`. Thank you once again for the suggestion.

My opinion is that the desktop file should be removed from `/etc` or the appropriate settings should be used so that it not enabled in Gnome. This is because gnome-tweaks does not show that blueman auto-starts and the only way a user can disable is to mess with config files.

If it was present in the autostart list and you could disable it in the GUI
1. Disabling would create a file in XDG_CONFIG_HOME and everything would work as expected.
2. It would be even better

I should not that birdtray, which is a similar type of utility does not ship a .desktop icon and enabling it in the autostart menu is pretty intuitive. Sure, tweaks is not shipped by default, but neither do tray icons.

**Comment by infirit** [comment]
> > You should find and have an option to disable blueman there.
> 
> It is not in tweaks nor in any of the directories for autostart (~/.config/autostart $XDG_CONFIG_HOME/autostart $XDG_CONFIG_DIRS/autostart /usr/share/gnome/autostart )
> 
> ```
> $ systemctl --user disable blueman-applet
> The unit files have no installation config (WantedBy=, RequiredBy=, Also=,
> Alias= settings in the [Install] section, and DefaultInstance= for template
> units). This means they are not meant to be enabled or disabled using systemctl.
> 
> Possible reasons for having this kind of units are:
> • A unit may be statically enabled by being symlinked from another unit's
>   .wants/ or .requires/ directory.
> • A unit's purpose may be to act as a helper for some other unit which has
>   a requirement dependency on it.
> • A unit may be started when needed via activation (socket, path, timer,
>   D-Bus, udev, scripted systemctl call, ...).
> • In case of template units, the unit is meant to be enabled with some
>   instance name specified.
> ```

Does it allow you to disable the service if you add
```
[Install]
WantedBy=multi-user.target
```

To `/usr/lib/systemd/user/blueman-applet.service` ?

**Comment by ispanos** [comment]
> Does it allow you to disable the service

I don't think the service is enabled. We figured out it was the .desktop file in `/etc/xdg/autostart`.

**Comment by infirit** [comment]
> > Does it allow you to disable the service
> 
> I don't think the service is enabled. We figured out it was the .desktop file in `/etc/xdg/autostart`.

I'm not so sure Gnome uses it at all. But you're welcome to ignore the suggestion and I'll be happy to close the issue as solved.

**Comment by ispanos** [comment]
> > > Does it allow you to disable the service
> > 
> > 
> > I don't think the service is enabled. We figured out it was the .desktop file in `/etc/xdg/autostart`.
> 
> I'm not so sure Gnome uses it at all. But you're welcome to ignore the suggestion and I'll be happy to close the issue as solved.

After adding

> ``` [Install]
> WantedBy=multi-user.target
> ```

I checked the status.

```
systemctl --user status blueman-applet
○ blueman-applet.service - Bluetooth management applet
     Loaded: loaded (/usr/lib/systemd/user/blueman-applet.service; disabled; preset: disabled)
    Drop-In: /usr/lib/systemd/user/service.d
             └─10-timeout-abort.conf
             /run/user/1000/systemd/user.control/blueman-applet.service.d
             └─50-CPUWeight.conf, 50-IOWeight.conf
     Active: inactive (dead)
     
     ....
```

> I'm not so sure Gnome uses it at all.

If you are referring to `/etc/xdg/autostart` it seems that it does use that path, because after removing the .desktop file blueman no longer starts.

**Comment by cschramm** [comment]
> Is that directory in `/etc/xdg/autostart` the proper place for blueman? I'm not asking for proof, if you are sure a yes would suffice.

Yes. We assume that users want to use blueman if it's installed and blueman-applet is vital for that, especially as it acts as the agent to BlueZ. It thus needs to get auto-started and the Desktop Application Autostart Specification is the absolute standard way to achieve that and supported by most desktop environments.

> A tray icon, especially when for some DE it is redundant, should be easy to disable. Tray icons are not even officially supported in Gnome.

blueman-applet is not about a tray icon. I agree that blueman is redundant for Gnome as typically gnome-bluetooth is in place and there is a chance that users of multiple desktop environments do not want to use blueman on Gnome, but there is just no `NotShowInIfUserSwitchesDesktopEnvironments=Gnome`.

If distributions or administrators decide to change defaults, they can always just remove or override the file, e.g. distributors might know that Gnome is not their default and thus probably not the only DE and opt in to `NotShowIn=Gnome`.

> My opinion is that the desktop file should be removed from `/etc` or the appropriate settings should be used so that it not enabled in Gnome. This is because gnome-tweaks does not show that blueman auto-starts and the only way a user can disable it is to mess with config files.

I'm surprised that Gnome seems to have dropped their autostart configuration completely (while still supporting `/etc/xdg/autostart`). That does not make any sense to me and is in contrast to other desktops (well, to be honest, it fits the overall picture :see_no_evil:). I agree that it's unfortunate that users have to mess with config files to disable blueman on Gnome. On the other hand it would be unfortunate that users have to set up their own auto-start mechanism and Gnome (without Tweaks) does not even provide a solution. Looks like we basically need to decide if we want to provide better defaults for users that want to use blueman on Gnome or users that want to use blueman on other desktops and not on Gnome.

**Comment by ispanos** [comment]
Anyway, the rest is up to you. I don't know how other desktops work. I barely know how gnome works since it does things in a way I cannot understand.

 I'm just in general someone who strongly believes in minimal system-wide configurations and easy (gui) ways to configure things for each user (on the same system) individually. Sane defaults are great, as long as they are easily configured on a per-user basis. I don't want to have to  manage multiple accounts on shared computers.

**Comment by infirit** [comment]
> I'm surprised that Gnome seems to have dropped their autostart configuration completely (while still supporting `/etc/xdg/autostart`). That does not make any sense to me and is in contrast to other desktops (well, to be honest, it fits the overall picture see_no_evil). I agree that it's unfortunate that users have to mess with config files to disable blueman on Gnome. On the other hand it would be unfortunate that users have to set up their own auto-start mechanism and Gnome (without Tweaks) does not even provide a solution. Looks like we basically need to decide if we want to provide better defaults for users that want to use blueman on Gnome or users that want to use blueman on other desktops and not on Gnome.

My gues is that it completely relies on systemd for this. There is a `xdg-desktop-autostart.target` which generates systemd units  for these autostart desktop files. See `systemctl --user list-dependencies xdg-desktop-autostart.target` which for me has blueman-applet listed.
```
xdg-desktop-autostart.target
● ├─app-bitwarden@autostart.service
○ ├─app-blueman@autostart.service
```

Now, with the below unit file I am able to control blueman-applet through `systemctl --user`. Now the question is, should we not install the autostart desktop file when systemd is detected or should we leave this to the distribution? One thing I worry about if we handle it is how we enforce systemd starting it for every user :thinking: .


```
[Unit]
Description=Bluetooth management applet
PartOf=graphical-session.target

[Service]
Type=dbus
BusName=org.blueman.Applet
ExecStart=/usr/bin/blueman-applet
Restart=on-failure

[Install]
WantedBy=default.target
```

**Comment by cschramm** [comment]
There's also no GUI for configuring the units, is there? But yes, that eases it from files to `systemctl --user`. Could be a point against the autostart file (on systemd systems). :thinking:

**Comment by infirit** [comment]
> There's also no GUI for configuring the units, is there? But yes, that eases it from files to `systemctl --user`.

Plasma has it's own service things which seems unrelated to systemd but it also does not have a UI for managing this afaics.

I just made a draft PR adding the bits above to the blueman-applet unit file #2105.

**Comment by ispanos** [comment]
This has been the most constructive conversation I've had about a problem for months. Thank you for engaging. Tag me if you want me to test something.

**Comment by cschramm** [comment]
I wanted to have a look at how birdtray implements autostart, however it does not seem to provide any such configuration. 🤔 It does ship a desktop file, typically to /use/share/applications. Possibly some configuration tools pick that up and offer to set up an autostart for it. An easy configuration solution could be to add and remove a link to such an application file in the autostart directory.

**Comment by ispanos** [comment]
> add and remove a link to such an application file in the autostart directory.

If it is in the home directory it would be even better.

**Comment by KAGEYAM4** [comment]
```
[Desktop Entry]
Name[en_GB]=NetworkManager Applet
Comment[en_GB]=Manage your network connections
Icon=nm-device-wireless
Exec=nm-applet
Terminal=false
Type=Application
NoDisplay=true
NotShowIn=KDE;GNOME;
X-GNOME-UsesNotifications=true
```

The above is of nm-applet. The purpose of installing nm-applet for user is similar to blueman. If a user runs Hyprland/Sway/Niri, they know that they will need to exec them by some means.

**Comment by jscott0** [comment]
Forgive me for not acknowledging the prior discussion, but comments on this issue have intertwined many different ideas and questions:
- how a user can disable auto-starting of an application, within GNOME in particular or in all desktop environments
- how—and if—Blueman can deliver a better default to serve GNOME users
- how to disable Blueman's services starting
- what disposition the Blueman applet should have towards Bluetooth being already turned on or not, via rfkill or otherwise
- what role systemd has in auto-starting applications versus the desktop environment
- what use case there is in someone having blueman-applet installed if they don't want to use it
- <i>et cetera</i>

Much of this discussion is out-of-date as well.

Anyway, I've been affected by this issue and for completeness I'm going to "start at the beginning" with a user experience story and suggested remedy. I'm using GNOME 3 on Debian Trixie. GNOME has GNOME Settings to manage Bluetooth and GNOME Shell will take care of notifications for pairing requests, file transfers, and the like. For typical and simple use cases, this should be good enough.
However, `blueman-manager` offers many "bells and whistles" and advanced options that are (intentionally) not exposed in GNOME's settings directly. GNOME's general style is to not overwhelm the user with settings. Thus, I sought to get the graphical Blueman Manager.
On Debian, all components of Blueman are shipped in a single binary package called `blueman`. (I have no stance at the moment whether this is good practice.) This package provides the graphical Blueman Manager that I've wanted so now I'm happy and can tinker with that, but this made a new problem.
The Blueman package installed `/etc/xdg/autostart/blueman.desktop` which causes `blueman-applet` to automatically start on login. GNOME Shell does not support the concept of tray applets at all anymore, but importantly this causes notifications to be shown by Blueman when GNOME already takes care of this. When my Bluetooth wireless mouse disconnects and reconnects, GNOME Shell now shows some noisy notifications from Blueman. But the point is, GNOME Shell already takes care of those functions that the Blueman Applet is designed for: basic notifications (for example with incoming file transfers), device management, and the like. In this regard GNOME Shell is special: someone with the Blueman Applet installed most likely has it because they want to use it, but it doesn't serve any purpose with GNOME because they've taken responsibility for the functions. It is effectively unconscionable that someone using GNOME Shell will want this applet running, at least by default, so a carve-out would be helpful.

On some (most?) setups, [systemd can take responsibility for getting the XDG autostart applications started](https://systemd.io/DESKTOP_ENVIRONMENTS/#xdg-autostart-integration). In that case, they respect settings such as `NotShowIn=GNOME` that are [standardized just for this purpose](https://specifications.freedesktop.org/autostart/latest/#id-1.3.4.4). I think modern versions of GNOME make systemd a mandatory dependency, and GNOME may not do autostarting of applications at its own initiative anymore, delegating this to systemd instead. If GNOME does take responsibility for autostarting applications anyway, it absolutely should respect this setting just the same, because the autostart specification explicitly says so.

In conclusion, it would be smart for the Blueman Applet to blacklist GNOME explicitly as a desktop environment it should automatically start in, and this can almost surely be obtained by adding the following line to `/etc/xdg/autostart/blueman.desktop`:

`NotShowIn=GNOME`

(Be careful: that says 'show', not 'shown'.)

This is semantic, and is exactly the recommended practice. I'll test it on my system momentarily, but please let me know if there's some detail I'm missing. Otherwise, I'll do some additional research to ensure soundness and then prepare a pull request. Thank you for your consideration

(Edit: a report of this being used effectively can be [found here](https://peterbabic.dev/blog/hide-blueman-applet-gnome-shell/), but note well: that user, like some folks here, are trying to accommodate having multiple desktop environments installed. In my experience I described, I use GNOME Shell exclusively, but the applet is still a mild disturbance at best and superfluous or troublesome at worst.)

**Comment by cschramm** [comment]
> In this regard GNOME Shell is special: someone with the Blueman Applet installed most likely has it because they want to use it, but it doesn't serve any purpose with GNOME because they've taken responsibility for the functions.

You seem to suggest that someone who wants to use blueman is simply mislead as GNOME Bluetooth has them covered. Why should they keep it then?

Also note that blueman-applet is a very vital part of blueman and it can only work properly if no other agent, like GNOME Bluetooth's, interferes. Someone who wants to use blueman should make sure that GNOME Bluetooth is disabled (however folks achieve that; maybe `NotShowIn=GNOME`? 🙈).

**Comment by doraeric** [comment]
I'm running into the same issue. I want to use the Bluetooth GUI, but I don't need it sitting in the system tray all the time—I’d rather just open it manually when I need it. I noticed the package installs `/etc/xdg/autostart/blueman.desktop`, so I followed the steps in this [link](https://unix.stackexchange.com/questions/563403/how-to-disable-etc-xdg-something) to prevent it from auto-starting in GNOME.
```sh
cp /etc/xdg/autostart/blueman.desktop ~/.config/autostart/blueman.desktop
```
and add this line
```
Hidden=true
```

I think it should be [GNOME Tweaks](https://gitlab.gnome.org/GNOME/gnome-tweaks)' responsibility to show system-wide autostart entries, but it doesn't.

**Comment by doraeric** [comment]
In my opinion, whether a GUI app starts automatically should be a per-user preference. Many applications like KeePassXC, MEGAsync, and Windscribe handle this by providing a toggle in their settings. When enabled, it creates an entry in `~/.config/autostart` instead of placing a file in `/etc/xdg/autostart`, which forces the behavior on every user at boot. I believe this per-user approach is a much better workflow.

**Comment by dnut** [comment]
This is an abuse of the XDG autostart system. Running an application should always be a distinct and deliberate user choice. Sneaking execution into the installation step without the user's explicit consent is unexpected and a blatant disregard of user autonomy. It is not reasonable to expect users to continuously monitor the list of running processes to detect if there is anything in there that they do not consent to. It is however perfectly reasonable to expect users to start the apps that they want to run.

**Comment by joezhouchenye** [comment]
Blueman is great since I have two adapters for my computer. The motherboard attached one doesn't work and I have to plug in another USB Bluetooth adpater. Blueman can help me choose which one to use while GNOME bluetooth cannot do this without manual configuration. However, I think many people prefer system integrated one unless they encounter some problem and have to install blueman? Therefore, it may be better to give the users an option whether to make blueman-applet autostart.

**Comment by ispanos** [comment]
I’m following up on my original report three years later. I now use neither Sway nor Blueman, but I dealt with a similar autostart problem recently and came back to this issue.

After rereading the discussion and the [XDG Autostart specification](https://specifications.freedesktop.org/autostart/latest/), I would revise part of my original position. `/etc/xdg/autostart/blueman.desktop` might be the correct place for a system-wide, enabled-by-default entry. Removing it would create a bootstrap problem and change the current default. The missing piece is an accessible per-user override.

Could Blueman add these settings?

1. `Start Blueman Applet at login`, enabled by default.
2. `Do not start in GNOME`, disabled by default.
3. `Do not start in KDE Plasma`, disabled by default.

These settings could manage `${XDG_CONFIG_HOME:-$HOME/.config}/autostart/blueman.desktop`:

- If autostart is disabled, write a same-named user entry containing `Hidden=true`.
- If autostart is enabled with no desktop exclusions, remove the user override and inherit the system-wide default.
- If either desktop exclusion is enabled, write a complete user entry with the appropriate `NotShowIn=GNOME;`, `NotShowIn=KDE;`, or combined value.

This would preserve the current behavior and address the concern that `blueman-applet` is a vital part of Blueman. I am not proposing that GNOME or KDE be excluded globally. A user who disables GNOME’s Bluetooth agent and wants Blueman there would retain the current default. Other users, including different users of the same computer, could choose independently.

This is close to @doraeric’s proposal above, but keeping the system entry as the default avoids changing existing behavior. It also means that neither the package nor Blueman needs to create a user file on first launch. The absence of a user override means “use the enabled system default.”

I also looked into whether package installation could create the user file for every account. There is no reliable, portable way to do that. Package installation does not know each user’s `XDG_CONFIG_HOME`, accounts may not exist locally or yet, and `/etc/skel` affects only subsequently created accounts. [Debian Policy also notes](https://www.debian.org/doc/debian-policy/ch-files.html#user-configuration-files-dotfiles) that there is no easy or necessarily desirable way to copy such defaults into existing accounts.

Finally, the Tweaks behavior now makes sense. [GNOME Tweaks currently builds its Startup Applications list from user autostart files only](https://github.com/GNOME/gnome-tweaks/blob/master/gtweak/tweaks/tweak_group_startup.py#L315-L319). A system-wide entry can therefore be honored by GNOME without appearing in Tweaks. That makes an autostart toggle in Blueman itself the most discoverable solution.

Note: this comment was written by Codex to help me convey my message clearly.

Edit:

> - If autostart is enabled with no desktop exclusions, remove the user override and inherit the system-wide default.

You may want to keep the override, so that the entry appears in gnome-tweaks.
